src/server/collections/commentLike-db.js

```diff
@@ -0,0 +1,153 @@
+} else {
+  url = protocol + '://' + host + ':' + port + '/' + name;
+}
+
+function dbError(err, res, db) {
+  if (err) {
+    res.status(404).send();
+    db.close();
+    return;
+  }
+}
+
+function update(videosId, commentId, votetype, res, req) {
+  let userId;
+
+  MongoClient.connect(url, (err, db) => {
+    if (err) {
+      throw err;
+    }
+    db.collection('videos').find({'videoId': videosId})
+      .toArray(function(err, items) {
+        dbError(err, res, db);
+        let VideoObject = items[0];
+        let token = req.get('Authorization');
+
+        db.collection('tokenDescriptors')
+          .find({'token': token})
+          .toArray(function(err, subitems) {
+            dbError(err, res, db);
+            if (items[0] === undefined) {
+              console.log('Not login', err);
+              res.status(404).send({'error': 'Not login sorry!'});
+              db.close();
+              return;
+            }
+            userId = subitems[0].userId;
+            let inputObj = {
+              'votetype': votetype,
+              'db': db,
+              'videosId': videosId,
+              'userId': userId,
+              'VideoObject': VideoObject,
+              'res': res,
+              'commentId': commentId,
+            };
+
+            commentLikeCallHandler(inputObj);
+          });
+      });
+  });
+}
+
+function countLikeAndDislikedNumber(commentUserArray) {
+  let likednumber = 0;
+  let dislikednumber = 0;
+
+  commentUserArray.forEach(function(element) {
+    if (element.liked === true) {
+      likednumber++;
+    }
+    if (element.disliked === true) {
+      dislikednumber++;
+    }
+  });
+  let result = {
+    'likedNumber': likednumber,
+    'dislikedNumber': dislikednumber,
+  };
+
+  return result;
+}
+
+function updateCommentsInfo(VideoObject, commentId, tempArray, db, videosId) {
+  VideoObject.commentInfos[commentId - 1].LikeStatus = tempArray;
+  db.collection('videos')
+    .update({'videoId': videosId},
+      {$set: {'commentInfos': VideoObject.commentInfos}});
+}
+
+function createObj(whetherFind, like, dislike, tempArray, userId) {
+  if (!whetherFind) {
+    let objInsert = {
+      'userId': userId.toString(),
+      'liked': like,
+      'disliked': dislike,
+    };
+
+    tempArray.push(objInsert);
+  }
+}
+
+function iterateArray(tempArray, changedValue, userId, target) {
+  for (let i = 0; i < tempArray.length; i++) {
+    if (tempArray[i].userId === userId.toString()) {
+      if (target === 'disliked') {
+        tempArray[i].disliked = changedValue;
+      } else if (target === 'liked') {
+        tempArray[i].liked = changedValue;
+      }
+      return true;
+    }
+  }
+}
+
+function sendResponse(tempArray, res, db) {
+  let sendObj = countLikeAndDislikedNumber(tempArray);
+
+  res.setHeader('content-type', 'application/json');
+  res.status(200).send(sendObj);
+  db.close();
+}
+
+function commentLikeCallHandler(obj) {
```

Comment:
it is too complex logic in a db file, it should just get some date and store/update or delete it. The logic should be in the controller